packages/extension-base/src/background/handlers/Extension.ts

```diff
@@ -95,6 +95,14 @@ export default class Extension {
     return { exportedJson: JSON.stringify(keyring.backupAccount(keyring.getPair(address), password)) };
   }
 
+  private async accountsBatchExport ({ addresses }: RequestAccountBatchExport): Promise<ResponseAccountExport> {
+    return {
+      exportedJson: JSON.stringify(
```

Comment:
As suggested on the ui repo, the ui-keyring should rather do the heavy-lifting. This means that wehn e.g. added in the apps UI, the logic is exactly the same. so move the `JSON.stringify` there.

Reply:
aye sir.

Reply:
My implementation doesn't add a new entry point for importing. It extends current `Restore account from backup JSON file`.
I suppose a new entry point would be confusing to users.

However, this implementation skips password confirming which used to be in importing process.
Thus I need your decision here, whether I should add a separate entry point for batch importing, or remove the password input from importing at all.

Reply:
Keep the password input. Keep as-is. We can just warp the keyring export with the new `jsonEncrypt` call and return an encypted value there. (Same for decrypt)

https://github.com/polkadot-js/common/blob/master/packages/util-crypto/src/json/encrypt.ts#L12

So for the contentType pass through `[batch-pkcs8]` (`pkcs8` is the account-only format)

(Data would be `stringToU8a(JSON.stringify(accounts))` in the keyring, decrypt would be the reverse)

Reply:
^^^ all the above in the UI PR in the keyring, so it returns a JSON encoded or a JSON decoded.

Reply:
Encrypted JSON brings up some problems.

- Currently the uploaded file is parsed right away to fetch account info via `jsonGetAccountInfo`. It's impossible with encrypted file content.
- And it can't be decided if an uploaded file requires password unless it's sent to ui-keyring for recovery, and that's too late.

We need to think about it very carefully.

Reply:
It is the same as the normal JSON files. Some meta information is not encrypted. Like address, name, tags. In this case just follow the same approach.

Reply:
cool